rpc/src/v1/impls/parity_accounts.rs

```diff
@@ -50,26 +50,33 @@ impl ParityAccountsClient {
 }
 
 impl ParityAccounts for ParityAccountsClient {
-	fn all_accounts_info(&self) -> Result<BTreeMap<RpcH160, BTreeMap<String, String>>, Error> {
+	fn all_accounts_info(&self) -> Result<BTreeMap<RpcH160, ExtAccountInfo>, Error> {
 		let store = self.account_provider()?;
 		let info = store.accounts_info().map_err(|e| errors::account("Could not fetch account info.", e))?;
 		let other = store.addresses_info();
 
-		Ok(info
-		   .into_iter()
-		   .chain(other.into_iter())
-		   .map(|(address, v)| {
-			   let mut m = map![
-				   "name".to_owned() => v.name,
-				   "meta".to_owned() => v.meta
-			   ];
-			   if let &Some(ref uuid) = &v.uuid {
-				   m.insert("uuid".to_owned(), format!("{}", uuid));
-			   }
-			   (address.into(), m)
-		   })
-		   .collect()
-		)
+		let account_iter = info
+			.into_iter()
+			.chain(other.into_iter())
+			.map(|(address, v)| (address.into(), ExtAccountInfo {
+				name: v.name,
+				meta: v.meta,
+				uuid: v.uuid.map(|uuid| uuid.to_string())
+			}));
+
+		let mut accounts: BTreeMap<RpcH160, ExtAccountInfo> = BTreeMap::new();
+
+		for (address, map) in account_iter {
+			match accounts.get(&address) {
+				// Skip if an account with UUID is already present.
+				Some(ref occupied) if occupied.uuid.is_some() => continue,
```

Comment:
`Entry` API could be used here to coalesce the "get or insert" logic, but I don't know if it would be cleaner...